<?xml version="1.0"?>
<launch>
  <!-- NOTE: This launch file is for DEBUG purposes only!
	     It is not intended as an entry point for running 
             simulations. 
             For running simulations use the taurob_bring_up 
             package or use the docker simulation enviroment 
             in taurob_gazebo/ros_container_simulation
  -->

  <arg
    name="laser_offset_rot"
    default="0.0" />

  <arg
    name="flipper_type"
    default="regular"
    doc="robot flipper type: ('regular', 'elongated')"/>

  <!-- Robot footprint config file path -->
  <arg
    name="file_path"
    value="$(find taurob_robots_description)/config/inspector_longarm"
    if="$(eval arg('flipper_type') == 'regular')" />

  <arg
    name="file_path"
    value="$(find taurob_robots_description)/config/inspector_longarm/flipper_elongated" 
    if="$(eval arg('flipper_type') == 'elongated')" />
  
  <!-- NOTE: For correctly setting up robot urdf based on 
       flipper type('regular', 'elongated').
       please set environment variable FLIPPER_TYPE
       e.g.: export ROBOT_MODEL_FLIPPER_TYPE="regular" 
             OR 
             export ROBOT_MODEL_FLIPPER_TYPE="elongated"
   -->
  <param
    name="robot_description"
    command="$(find xacro)/xacro '$(find taurob_robots_description)/urdf/inspector_longarm/model_with_arm.urdf.xacro'" />

  <!-- send fake joint values -->
  <node name="joint_state_publisher_gui" pkg="joint_state_publisher_gui" type="joint_state_publisher_gui">
     <param name="use_gui" value="1"/>
     <rosparam>
    zeros:
        arm_joint_1: 2.01
        arm_joint_2: 3.375
        arm_joint_3: 0.15
        arm_joint_4: 2.95
    </rosparam>
  </node>

  <!-- Publish robot global/local footprint -->
  <node
      name="robot_footprint_publisher"
      pkg="taurob_robots_description"
      type="robot_footprint_publisher.py"
      output="screen">
      <rosparam
      command="load"
      file="$(arg file_path)/global_footprint.yaml"
      ns="footprint_global"/>
      <rosparam
      command="load"
      file="$(arg file_path)/local_footprint.yaml"
      ns="footprint_local"/>
  </node>

  <!-- Combine joint values -->
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"/>

  <!-- Show in Rviz   -->
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find taurob_robots_description)/launch/default.rviz"/>

</launch>
